File: Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Orders/Queries/OrderLookupCriteria.cs
namespace Tuckshop.Models.Orders.Queries
{
  using System;
  using Tuckshop.Models.Orders.Enums;

  /// <summary>
  /// The criteria for querying orders
  /// </summary>
  public class OrderLookupCriteria
  {
    /// <summary>
    /// Gets or sets the Order Status filter value
    /// </summary>
    public OrderStatus? OrderStatus { get; set; }

    /// <summary>
    /// Gets or sets the Start Date filter value
    /// </summary>
    public DateTime? StartDate { get; set; }

    /// <summary>
    /// Gets or sets the End Date filter value
    /// </summary>
    public DateTime? EndDate { get; set; }
  }
}

File: Client/Tuckshop.WebApp/src/Domain/Models/Orders/Queries/OrderLookupCriteria.ts
import { Attributes, Misc, Validation, ValueObject } from '@singularsystems/neo-core';
import { OrderStatus } from '../Enums/OrderStatus';

export default class OrderLookupCriteria extends ValueObject {

    constructor() {
        super();
        this.makeObservable();
    }

    public orderStatus: OrderStatus | null = null;

    @Attributes.Date(Misc.TimeZoneFormat.None)
    public startDate: Date | null = null;

    @Attributes.Date(Misc.TimeZoneFormat.None)
    public endDate: Date | null = null;

    public customerName: String | null = null;

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<OrderLookupCriteria>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        return "Order Lookup Criteria";
    }
}